src/main/java/org/myrobotlab/service/interfaces/ServoControl.java

```diff
@@ -67,7 +67,7 @@
    * @throws Exception
    *           e
    */
-  void attachServoController(ServoController controller) throws Exception;
+  void attachServoController(Attachable controller) throws Exception;

```

Comment:
Ok with this change, however, it seems like you should only be allowed to pass in a ServoController here...  which also says that "Simulator" and "ServoController" should be the same interface for purproses of a JME3 engine?

Reply:
I did not want to bloat Simulators by making them implement every possible controller we want to virtualize ... I also didn't want to crowd our Services with "Implementation" services who's sole purpose is to provide a ServoController for a very specific Simulator ...
E.g. I don't thing we need a full blown JMEServoController service ... additionally I don't want the Simulator itself to implement every sing method specified in the ServoController interface.

The attach routes based on type to the appropriate details to attach a simulator to a servo (through a lightweight JmeServoController which is just a class not a service in the org.myrobotlab.jme package space)